routing/dht/routing.go

```diff
@@ -84,7 +84,10 @@ func (dht *IpfsDHT) GetValue(ctx context.Context, key key.Key) ([]byte, error) {
 	ctx, cancel := context.WithTimeout(ctx, time.Minute)
 	defer cancel()
 
-	vals, err := dht.GetValues(ctx, key, 16)
+	// retrieve a majority of the expected record count
+	majority := (KValue / 2) + 1
```

Comment:
> I dont think the number of records we require should be greater than the number of records that a single store operation puts.

Yep, agreed, feel free to bump that up!

> That is the reason for a lot of the slowness, we put out 10 records, and then try to get 16 from the network.

No, the slowness is that the DHT currently _does not_ exhaust the query set correctly-- it considers nodes NOT on the gradient path. It ends up talking to way more nodes than it should before terminating a query.

Do you understand what i'm saying here? o/ 

> if a node requests K, and receives values A B B A C B, and A is the most valid record, then the node making the query will 'correct' all the nodes who did not send A by sending them a STORE rpc with the value A. Because of this, the probabilities of getting the freshest record are greatly improved.

of course, that is a very useful mechanic. However it does not _at all_ guarantee that -- after natural churn in the network -- nodes will actually be able to find the freshest records in the first place! And this wont be immediately apparent, it will be an insidious bug that harms people over time but is tricky to reproduce. 

For IPNS, _**Safety**_ is critical.  _**Liveness**_.  is achieved through optimization, but never at the cost of safety. 

---

please go back and reread my comments -- sp https://github.com/ipfs/go-ipfs/pull/1887/files#r43575580 --, and reread the relevant papers (like s/kademlia http://www.tm.uka.de/doc/SKademlia_2007.pdf (the query resolution parts) and the survey papers on DHTs). this is not an argument that is useful to have without a strong grounding in the actual probabilistic models these types of systems assume, and trying to explain it in this issue is not going to do even a tenth of a job as good as those papers.